extensions/functions_datetime.yaml

```diff
@@ -265,3 +340,274 @@ scalar_functions:
           - name: y
             value: interval_year
         return: boolean
+  -
+    name: "assume_timezone"
+    description: >-
+      Convert local timestamp to UTC-relative timestamp_tz using given local time's timezone.
+
+      Timezone strings must be as defined by IANA timezone database (https://www.iana.org/time-zones).
+      Examples: "Pacific/Marquesas", "Etc/GMT+1".
+      If timezone is invalid an error is thrown.
+    impls:
+      - args:
+          - name: x
+            value: timestamp
+          - name: timezone
+            description: Timezone string from IANA tzdb.
+            value: string
+        return: timestamp_tz
+      - args:
+          - name: x
+            value: date
```

Comment:
I assume the returned timestamp_tz is for midnight but it might be nice to clarify in the above description.

Reply:
That'll work I suppose.  The argument description isn't really the first place I'd think to look for that information.  However, the spec format doesn't really have any good spot for implementation-specific notes.